packages/app/src/cli/models/app/loader.ts

```diff
@@ -313,36 +355,30 @@ class AppLoader {
     extensionConfigPaths.push(`!${joinPath(appDirectory, '**/node_modules/**')}`)
     const configPaths = await glob(extensionConfigPaths)
 
-    const extensions = configPaths.map(async (configurationPath) => {
+    const extensionPromises = configPaths.map(async (configurationPath) => {
       const directory = dirname(configurationPath)
-      const specification = await findSpecificationForConfig(
-        this.specifications,
-        configurationPath,
-        this.abortOrReport.bind(this),
-      )
-
-      if (!specification) return
 
-      const configuration = await this.parseConfigurationFile(specification.schema, configurationPath)
-      const entryPath = await this.findEntryPath(directory, specification)
-
-      const extensionInstance = new ExtensionInstance({
-        configuration,
-        configurationPath,
-        entryPath,
-        directory,
-        specification,
-      })
-
-      const validateResult = await extensionInstance.validate()
-      if (validateResult.isErr()) {
-        this.abortOrReport(outputContent`\n${validateResult.error}`, undefined, configurationPath)
+      const fileContent = await readFile(configurationPath)
+      const obj = decodeToml(fileContent)
+      const {extensions, type} = ExtensionsArraySchema.parse(obj)
+
+      if (extensions) {
+        // If the extension is an array, it's a unified toml file.
+        // Parse all extensions by merging each extension config with the global unified configuration.
+        const configuration = await this.parseConfigurationFile(UnifiedSchema, configurationPath)
+        const extensionsPromises = configuration.extensions.map(async (extensionConfig) => {
```

Comment:
nit: you have an `extensionsPromises` variable inside your `extensionPromises` map which is a tad hard to read. Since `extensionsPromises` is using `createExtensionInstance` could we rename it to `extensionsInstances` or `extensionsInstancesPromises`?